nucleus/hk2/hk2-config/src/main/java/org/jvnet/hk2/config/WriteableView.java

```diff
@@ -317,16 +318,16 @@ public synchronized boolean canCommit(Transaction t) throws TransactionFailure {
         return currentTx==t;
     }
 
-    private void handleValidationException(Set constraintViolations) throws ConstraintViolationException {
+    private void handleValidationException(Set<? extends ConstraintViolation<?>> constraintViolations) throws ConstraintViolationException {
 
         if (constraintViolations != null && !constraintViolations.isEmpty()) {
-            Iterator<ConstraintViolation<ConfigBeanProxy>> it = constraintViolations.iterator();
```

Comment:
NB. Filling in generics the `ConfigBeanProxy` has shown to be not correct. While this might be the actual type the generics along the path to this point do not actually provide this guarantee.